network-api/networkapi/buyersguide/management/commands/migrate_products.py

```diff
@@ -0,0 +1,237 @@
+                'user_friendly_privacy_policy', 'show_ding_for_minimum_security_standards',
+                'meets_minimum_security_standards', 'uses_encryption',
+                'uses_encryption_helptext', 'security_updates', 'security_updates_helptext',
+                'strong_password', 'strong_password_helptext', 'manage_vulnerabilities',
+                'manage_vulnerabilities_helptext', 'privacy_policy', 'privacy_policy_helptext',
+                'phone_number', 'live_chat', 'email', 'twitter'
+            ]
+
+            # Add custom fields based on the product type
+            if isinstance(product, SoftwareProduct):
+                # Handle exceptions first. Then add fields to loop through.
+                new_product_page.medical_privacy_compliant = (
+                    False if not product.medical_privacy_compliant else product.medical_privacy_compliant
+                )
+                new_product_page.easy_to_learn_and_use = (
+                    False if not product.easy_to_learn_and_use else product.easy_to_learn_and_use
+                )
+
+                fields = common_fields + [
+                    'handles_recordings_how', 'recording_alert', 'recording_alert_helptext',
+                    'medical_privacy_compliant_helptext', 'host_controls', 'easy_to_learn_and_use_helptext'
+                ]
+            elif isinstance(product, GeneralProduct):
+                fields = common_fields + [
+                    'camera_device', 'camera_app', 'microphone_device', 'microphone_app',
+                    'location_device', 'location_app', 'personal_data_collected',
+                    'biometric_data_collected', 'social_data_collected',
+                    'how_can_you_control_your_data', 'data_control_policy_is_bad',
+                    'track_record_choices', 'company_track_record', 'track_record_is_bad',
+                    'track_record_details', 'offline_capable', 'offline_use_description',
+                    'uses_ai', 'ai_uses_personal_data', 'ai_is_transparent', 'ai_helptext'
+                ]
+
+            self.debug_print("\tSetting fields:")
+            for field in fields:
+                # Loop through every field for this product and copy the value
+                # from the Product model to the Page model.
+                self.debug_print("\t\t", field, " as ", getattr(product, field))
+                setattr(new_product_page, field, getattr(product, field))
+
+            # Get the image file field, and convert it into a WagtailImage object
+            # I don't know why but convert_pni_file_to_product_page_image() doesn't
+            # seem to work. I tinkered with it but ultimately gave up
+            # because this a temporary management command anyway and should ideally
+            # be used once in production, and code deleted afterwards.
+            mime = MimeTypes()
+            mime_type = mime.guess_type(product.image.file.name)  # -> ('image/jpeg', None)
```

Comment:
This causes an error locally: if I copy the prod db, run `inv migrate` and then `inv manage migrate_products`, there is apparently no `product.image.file` and so this breaks.... this probably needs some deeper investigation:
```
Creating foundationmozillaorg_backend_run ... done
Traceback (most recent call last):
  File "network-api/manage.py", line 34, in <module>
    execute_from_command_line(sys.argv)
  File "/app/dockerpythonvenv/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/app/dockerpythonvenv/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/app/dockerpythonvenv/lib/python3.7/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/app/dockerpythonvenv/lib/python3.7/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/app/network-api/networkapi/buyersguide/management/commands/migrate_products.py", line 133, in handle
    mime_type = mime.guess_type(product.image.file.name)  # -> ('image/jpeg', None)
  File "/app/dockerpythonvenv/lib/python3.7/site-packages/django/db/models/fields/files.py", line 41, in _get_file
    self._require_file()
  File "/app/dockerpythonvenv/lib/python3.7/site-packages/django/db/models/fields/files.py", line 38, in _require_file
    raise ValueError("The '%s' attribute has no file associated with it." % self.field.name)
ValueError: The 'image' attribute has no file associated with it.
```

Reply:
We'll need to check if there is an image and a file. If not, don't try to copy it. Ill add that conditional in shortly.